n3fit/src/n3fit/stopping.py

```diff
@@ -438,49 +336,69 @@ def __init__(
         threshold_chi2=10.0,
         dont_stop=False,
     ):
+        self._pdf_models = pdf_models
+
         # Save the validation object
         self._validation = validation_model
 
         # Create the History object
         tr_ndata, vl_ndata, pos_sets = parse_ndata(all_data_dicts)
-        self._history = FitHistory(pdf_models, tr_ndata, vl_ndata)
+        self._history = FitHistory(tr_ndata, vl_ndata)
 
         # And the positivity checker
         self._positivity = Positivity(threshold_positivity, pos_sets)
 
         # Initialize internal variables for the stopping
-        self.n_replicas = len(pdf_models)
-        self.threshold_chi2 = threshold_chi2
-        self.stopping_degree = np.zeros(self.n_replicas, dtype=int)
-        self.count = np.zeros(self.n_replicas, dtype=int)
-
-        self.dont_stop = dont_stop
-        self.stop_now = False
-        self.stopping_patience = stopping_patience
-        self.total_epochs = total_epochs
+        self._n_replicas = len(pdf_models)
+        self._threshold_chi2 = threshold_chi2
+        self._stopping_degrees = np.zeros(self._n_replicas, dtype=int)
+        self._counts = np.zeros(self._n_replicas, dtype=int)
+
+        self._dont_stop = dont_stop
+        self._stop_now = False
+        self._stopping_patience = stopping_patience
+        self._total_epochs = total_epochs
+
+        self._stop_epochs = [total_epochs - 1] * self._n_replicas
+        self._best_epochs = [None] * self._n_replicas
+        self.positivity_statusses = [POS_BAD] * self._n_replicas
```

Comment:
```suggestion
        self.positivity_statuses = [POS_BAD] * self._n_replicas
```
Also in the subsequent occurrences.